### PR TITLE
Validate user-provided entity versions are url safe

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
Ensures that user-provided entity version are url safe since they are used http paths to manipulate the entities and can lead to 404s when they include slashes.

Execution names, which are also user-provided and used in http paths are already [constrained](https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/impl/validation/execution_validator.go#L22) to a smaller subset of allowable chars

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Encountered with a user

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/3427

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
cc @cosmicBboy
